test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/RequestSizeLimitResourceFilterTest.cs

```diff
@@ -0,0 +1,162 @@
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Logging.Testing;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.Internal
+{
+    public class RequestSizeLimitResourceFilterTest
+    {
+        [Fact]
+        public void SetsMaxRequestBodySize()
+        {
+            // Arrange
+            var requestSizeLimitResoureFilter = new RequestSizeLimitResourceFilter(NullLoggerFactory.Instance);
+            requestSizeLimitResoureFilter.Bytes = 12345;
+            var resourceExecutingContext = CreateResourceExecutingContext(new IFilterMetadata[] { requestSizeLimitResoureFilter });
+
+            var httpMaxRequestBodySize = new TestHttpMaxRequestBodySizeFeature();
+            resourceExecutingContext.HttpContext.Features.Set<IHttpMaxRequestBodySizeFeature>(httpMaxRequestBodySize);
+
+            // Act
+            requestSizeLimitResoureFilter.OnResourceExecuting(resourceExecutingContext);
+
+            // Assert
+            Assert.Equal(12345, httpMaxRequestBodySize.MaxRequestBodySize);
+        }
+
+        [Fact]
+        public void SkipsWhenOverridden()
+        {
+            // Arrange
+            var requestSizeLimitResoureFilter = new RequestSizeLimitResourceFilter(NullLoggerFactory.Instance);
+            requestSizeLimitResoureFilter.Bytes = 12345;
+            var disableRequestSizeLimitResourceFilter = new DisableRequestSizeLimitResourceFilter(NullLoggerFactory.Instance);
+            var resourceExecutingContext = CreateResourceExecutingContext(new IFilterMetadata[] { requestSizeLimitResoureFilter, disableRequestSizeLimitResourceFilter});
+
+            var httpMaxRequestBodySize = new TestHttpMaxRequestBodySizeFeature();
+            resourceExecutingContext.HttpContext.Features.Set<IHttpMaxRequestBodySizeFeature>(httpMaxRequestBodySize);
+
+            // Act
+            requestSizeLimitResoureFilter.OnResourceExecuting(resourceExecutingContext);
+
+            // Assert
+            Assert.Null(httpMaxRequestBodySize.MaxRequestBodySize);
+        }
+
+        [Fact]
+        public void LogsFeatureNotFound()
+        {
+            // Arrange
+            var sink = new TestSink();
+            var loggerFactory = new TestLoggerFactory(sink, enabled: true);
+
+            var requestSizeLimitResoureFilter = new RequestSizeLimitResourceFilter(loggerFactory);
+            requestSizeLimitResoureFilter.Bytes = 12345;
+            var resourceExecutingContext = CreateResourceExecutingContext(new IFilterMetadata[] { requestSizeLimitResoureFilter });
+
+            // Act
+            requestSizeLimitResoureFilter.OnResourceExecuting(resourceExecutingContext);
+
+            // Assert
+            Assert.Single(sink.Writes);
```

Comment:
`var write = Assert.Single(sink.Writes);` then use `write` to simplify the next line. Same for other new tests.